File: 169_majority_element/Solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int votes = 0, x = 0;
        for (int num : nums) {
            if (votes == 0) x = num;
            if (x == num) votes++;
            else votes--;
        }

        return x;
    }

    public static void main(String[] args) {
        int[] nums = new int[]{1, 2, 2, 2, 3, 3};
        System.out.println(new Solution().majorityElement(nums));
    }
}

File: 169_majority_element/Solution.py
class Solution:
    def majorityElement(self, nums: list) -> int:
        votes, x = 0, 0
        for num in nums:
            if votes == 0: x = num
            if x == num: votes += 1
            else: votes -= 1

        return x


if __name__ == '__main__':
    nums = [1, 2, 2, 3, 2, 3, 2]
    print(Solution().majorityElement(nums))

